jsx/App/locale/TranslatableText.jsx

```diff
@@ -9,6 +9,11 @@ export const TranslatableText = ({ dictionary }) => {
   const { locale } = useContext(LocaleContext);
   if (dictionary[locale] === undefined) {
     console.error(`[TranslatableText] No translation for ${locale} in provided dictionary`);
+    // Try to return default language, if that is undefined, then just return
+    // whatever the first element of the dictionary is.
+    if (dictionary[DEFAULT_LOCALE] === undefined) {
+      return dictionary[Object.keys(dictionary)[0]];
```

Comment:
Using property order to decide which language to use might give unexpected results, since the order of keys on the object is not guaranteed in javascript. So for the COFAN example, if DEFAULT=ES and the dictionary defined for the glossary page was ` { "ai": ..., "en": ...., "fr": ... }` then this might render it in ai,en, or fr - depending on the browser.

I can think of a few other ways to solve the COFAN [Glossary page](https://github.com/Cofan-ALDP/LingView/pull/62/commits/0634604f0dd9e1c9aa5bd365f822dafff0c5ed17#) issue that don't depend on property order -
1) Since seemingly none of the text on the glossary page should be translated for COFAN project, why not just skip the `TranslatableText` component altogether for this page. (this seems preferable to me?)
2) If some things on the glossary should be translatable (such as title `Glossing Conventions` or title headings) then just use TranslatableText component for each of those small bits of text (and include translations to the default language- spanish--  for these).
3) Change this code to only have this special behavior if there is only **one** key on the object (`Object.keys(dictionary).length === 1`)

just my 2 cents! now I see you already merged this but I'll leave this comment anyway. I think this will be one of the most important components to make the development process for folks smooth, so just thinking ahead to what bugs might come up for people.

Reply:
The key idea behind this feature is that real text, even if it's in the wrong language, is always preferable to returning "undefined". To make it deterministic, we can replace the default-language property with a ranked list of all the languages, like [ENGLISH, SPANISH, FRENCH]. Then when getting the text, we first look for the user's chosen language if it exists, followed by English, then Spanish, and finally French if no other language is present. Do you think it's worth that change? 

We may want to translate the glossary page in the future, and other LingView users might as well, so I want to keep using TranslatableText for that page. 

Smaller translatable components is conceptually a very nice approach, but I worry about practicality. I want any user of LingView to be able to input or edit translations, even if they have minimal coding background. Scott suggested asking a couple members of the Cofan community to translate, for example, and while these research assistants are proficient in ELAN software and comfortable with spreadsheets, html/js/jsx might be completely new to them. I might be wrong, but I think editing the text of existing TranslatableText components is much more approachable than creating new TranslatableText components. Do you agree?

Reply:
Thanks for this context Kalinda! That all makes sense.

interesting idea about a ranking strategy for preferred languages. With the little context I have, I would say it is probably an overoptimization at this stage-- better to wait and see how people use the multilingualism feature first.

> I might be wrong, but I think editing the text of existing TranslatableText components is much more approachable than creating new TranslatableText components. Do you agree?

yeah I agree! thanks for highlighting that. editing the JSX in an existing dictionary object is more beginner-friendly because they don't have to touch the React component at all or really understand how it works.